core/src/main/scala/kafka/server/FetcherEventBus.scala

```diff
@@ -0,0 +1,177 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka.server
+
+import kafka.utils.CoreUtils.inLock
+import org.apache.kafka.common.utils.Time
+import java.util.{Comparator, PriorityQueue}
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.locks.{Condition, Lock, ReentrantLock}
+
+import kafka.utils.DelayedItem
+
+import scala.util.control.Breaks.{break, breakable}
+
+
+class QueuedFetcherEvent(val event: FetcherEvent,
+  val enqueueTimeMs: Long) extends Comparable[QueuedFetcherEvent] {
+  override def compareTo(other: QueuedFetcherEvent): Int = event.compareTo(other.event)
+}
+
+/**
+ * The SimpleScheduler is not thread safe
+ */
+class SimpleScheduler[T <: DelayedItem] {
+  private val delayedQueue = new PriorityQueue[T](new Comparator[T]() {
+    override def compare(t1: T, t2: T): Int = {
+      // here we use natural ordering so that events with the earliest due time can be checked first
+      t1.compareTo(t2)
```

Comment:
Will Comparator.naturalOrder() work here?

Reply:
Yes, thanks for the advice, and I'll use it.